bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -256,53 +233,60 @@ public BigtableSession(BigtableOptions options, @Nullable ExecutorService batchP
       throw new IllegalStateException("Neither Jetty ALPN nor OpenSSL via " +
           "netty-tcnative were properly configured.");
     }
-    if (batchPool == null) {
-      this.terminateBatchPool = true;
-      this.batchPool = createDefaultBatchPool();
-    } else {
-      this.terminateBatchPool = false;
-      this.batchPool = batchPool;
-    }
     this.options = options;
     Builder<HeaderInterceptor> headerInterceptorBuilder = new ImmutableList.Builder<>();
+    ExecutorService initializerExecutor =
+        Executors.newCachedThreadPool(ThreadPoolUtil.createThreadFactory("BigtableSessionInit"));
 
-    // Looking up Credentials takes time. Creating the retry executor and the EventLoopGroup don't
-    // take as long, but still take time. Get the credentials on one thread, and start up the elg
-    // and scheduledRetries thread pools on another thread.
-
-    Future<Void> credentialsFuture =
-        this.batchPool.submit(createCredentialsCallback(headerInterceptorBuilder));
-    this.elg = (elg == null) ? createDefaultEventLoopGroup() : elg;
-
-    this.scheduledRetries =
-        (scheduledRetries == null) ? createDefaultRetryExecutor() : scheduledRetries;
+    try {
+      // Looking up Credentials takes time. Creating the retry executor and the EventLoopGroup don't
+      // take as long, but still take time. Get the credentials on one thread, and start up the elg
+      // and scheduledRetries thread pools on another thread.
+      Future<Void> credentialsFuture =
+          initializerExecutor.submit(createCredentialsCallback(headerInterceptorBuilder));
+
+      BigtableSessionThreadPools.Builder threadPoolsBuilder =
+          new BigtableSessionThreadPools.Builder();
+      if (elg != null) {
```

Comment:
We're going to deprecate this constructor, and close the elg and retry pool right away.  Users should not  be using this.